tablecodec/tablecodec.go

```diff
@@ -388,10 +388,12 @@ func Unflatten(datum types.Datum, ft *types.FieldType, inIndex bool) (types.Datu
 	case mysql.TypeFloat:
 		datum.SetFloat32(float32(datum.GetFloat64()))
 		return datum, nil
+	case mysql.TypeVarchar, mysql.TypeString:
+		datum.SetString(datum.GetString())
```

Comment:
`TypeString` and `TypeVarchar` not always be string type.
`[]byte` and `string` should be used interchangeable.
The difference is charset.
The binary types have `binary` charset, string types have `utf8` charset.